gst/nnstreamer/nnstreamer_plugin_api_util_impl.c

```diff
@@ -1430,3 +1430,50 @@ nnstreamer_version_fetch (guint * major, guint * minor, guint * micro)
   if (micro)
     *micro = (NNSTREAMER_VERSION_MICRO);
 }
+
+/**
+ * @brief Compare dimension strings
+ * @return TRUE if equal, FALSE if given dimension strings are invalid or not equal.
+ */
+gboolean
+gst_tensors_dimension_string_equal (const gchar * dimstr1,
+    const gchar * dimstr2)
+{
+  tensor_dim dim1, dim2;
+  int rank1, rank2, i, j, num_tensors1, num_tensors2;
+  gchar **strv1;
+  gchar **strv2;
+  gboolean res = TRUE;
+
+  strv1 = g_strsplit (dimstr1, ",", -1);
+  strv2 = g_strsplit (dimstr2, ",", -1);
+
+  num_tensors1 = g_strv_length (strv1);
+  num_tensors2 = g_strv_length (strv2);
+
+  if (num_tensors1 != num_tensors2) {
+    res = FALSE;
+  } else {
+    for (i = 0; i < num_tensors1; i++) {
+      rank1 = gst_tensor_parse_dimension (strv1[i], dim1);
+      rank2 = gst_tensor_parse_dimension (strv2[i], dim2);
+
+      if (!rank1 || !rank2) {
+        res = FALSE;
+        break;
+      }
+
+      for (j = 0; j < NNS_TENSOR_RANK_LIMIT; j++) {
+        if (dim1[j] != dim2[j]) {
+          res = FALSE;
+          break;
```

Comment:
I think it should exit to line 1475 when break .
But. This code goes on line 1472.

Reply:
Added condition to outer for loop.
Thank you :)

Reply:
Thank you